pipeline_dp/dp_engine.py

```diff
@@ -52,6 +52,13 @@ def aggregate(self, col, params: AggregateParams,
         """
         if params is None:
             return None
+
+        with self._budget_accountant.scope(weight=params.budget_weight):
```

Comment:
I like an idea to create internal aggregate instead of having with on the entire function!